custom-metrics-stackdriver-adapter/examples/prometheus-to-sd/Dockerfile

```diff
@@ -12,10 +12,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.12.5
-ADD . /go/src/prometheus-dummy-exporter
-RUN go get github.com/prometheus/client_golang/prometheus
+FROM golang:1.16-alpine AS builder
+
+WORKDIR /app
+
+COPY go.mod ./
+COPY go.sum ./
+RUN go mod download
```

Comment:
do we want to have a vendor folder instead of running go mod download when building?

Reply:
this is a intermediate step which is going to be discarded anyways, and the base image in the last stage `gcr.io/distroless/static` makes sure our final image only contains an executable file